src/coffee/import.coffee

```diff
@@ -362,14 +370,20 @@ class Import
         when 'removeVariant' then @allowRemovalOfVariants
         else throw Error "The action '#{action.action}' is not supported. Please contact the commercetools support team!"
 
+    updateRequest = filtered.getUpdatePayload()
+
+    if publish
+      updateRequest.actions.push
+        action: 'publish'
+
     if @dryRun
       if filtered.shouldUpdate()
-        Promise.resolve "[row #{rowIndex}] DRY-RUN - updates for #{existingProduct.id}:\n#{_.prettify filtered.getUpdatePayload()}"
+        Promise.resolve "[row #{rowIndex}] DRY-RUN - updates for #{existingProduct.id}:\n#{_.prettify updateRequest}"
       else
         Promise.resolve "[row #{rowIndex}] DRY-RUN - nothing to update."
     else
       if filtered.shouldUpdate()
-        @client.products.byId(filtered.getUpdateId()).update(filtered.getUpdatePayload())
+        @client.products.byId(filtered.getUpdateId()).update(updateRequest)
         .then (result) =>
           @publishProduct(result.body, rowIndex)
```

Comment:
I am not sure there is any need for this again, since we are already adding the update action to publish above.

You can add an `if` condition not to call this method if publish product is already called.